### PR TITLE
String Interpolation 

### PR DESCRIPTION
**Pull request by jash11**
Based on #6339

### CODE REVIEW

File: std/experimental/interpolate.d [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
+}
+
+/**
+Interpolate an expression. `str` starts with an expression to interpolate.
+*/
+private string interpExpression(string str, string file, size_t line) pure @safe
+{
+    if (str[0] == '(')
+    {
+        uint depth = 1;
```
Comment (jash11) [inline std/experimental/interpolate.d]: Would it make more sense to use `size_t` or `ptrdiff_t` instead of `uint` here?

Comment (jash11) [reply std/experimental/interpolate.d]: nvm, I was confusing depth with i

**Comment by jash11** [comment]
Okay, this is ready for review now

**Comment by thewilsonator** [comment]
This needs a changelog for this to go in.

**Comment by wilzbach** [comment]
Also, please mind the style failures on Buildkite.

**Comment by jash11** [comment]
> I'm still highly in favor of a language solution as this is just too much too type for such a basic feature (closing parentheses and opening them is less characters!), the compiler implementation will be faster and I would have a hard time explaining to beginners why we have to use a `mixin` for this rudimentary feature.

@wilzbach I 100% agree with you, and I am also in favor of string interp as a language feature (Although I was getting the impression that I was the only one, because everyone was so strongly against it). Maybe that's not true? Is there still some hope in a language solution to interp 😄.

**Comment by wilzbach** [comment]
Hehe, the DMD PR has the highest voting count ever reached for a DMD PR and a lot of people are in favor of this (except for Andrei, of course).

**Comment by jash11** [comment]
I will close this then, because it is better to fight for that cause than for this.

**Pull request closed without merging**